src/core/transpile.jl

```diff
@@ -1592,3 +1592,128 @@ function update_qubit_mapping!(qubit_mapping::Dict{Int,Int}, gate::Swap)
     qubit_mapping[connected_qubits[1]] = outlet_qubit_1
     qubit_mapping[connected_qubits[2]] = outlet_qubit_2
 end
+
+struct SimplifyTrivialGatesTranspiler<:Transpiler 
+    atol::Real
+end
+
+SimplifyTrivialGatesTranspiler()=SimplifyTrivialGatesTranspiler(1e-6)
+
+function is_trivial_gate(gate::AbstractGate;atol=1e-6)::Bool
+    
+    params=get_gate_parameters(gate)
```

Comment:
I'm confuse, `get_gate_parameter` is the getter for the concrete gate types: 
`get_gate_parameters(gate::AbstractGate)
get_gate_parameters(gate::UnionMovedGates)
get_gate_parameters(gate::Rotation)
get_gate_parameters(gate::RotationX)
get_gate_parameters(gate::RotationY)
get_gate_parameters(gate::PhaseShift)
get_gate_parameters(gate::Universal)`

Reply:
`get_gate_parameters` returns the parameters as needed by `AnyonQPU` (coupling to AnyonQPU) and is defined as `get_gate_parameters(gate::AbstractGate)`. It is part of AnyonQPU and changes with AnyonQPU (if we want to change the name of a parameter in AnyonQPU, then this unrelated code will break. If you already have a type switch, you can use `get_theta` instead of `get_parameters()["theta"]`.

Reply:
As discussed on teams, we are going to leave it as is for now, until we figure out the best way to handle this.